src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs

```diff
@@ -602,21 +602,44 @@ public static string Write(OpenIdConnectConfiguration OpenIdConnectConfiguration
         {
             using (MemoryStream memoryStream = new MemoryStream())
             {
-                Utf8JsonWriter writer = null;
-                try
-                {
-                    writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping });
-                    Write(ref writer, OpenIdConnectConfiguration);
-                    writer.Flush();
-                    return Encoding.UTF8.GetString(memoryStream.GetBuffer(), 0, (int)memoryStream.Length);
-                }
-                finally
-                {
-                    writer?.Dispose();
-                }
+                return Write(OpenIdConnectConfiguration, memoryStream);
             }
         }
 
+        public static string Write(OpenIdConnectConfiguration OpenIdConnectConfiguration, Stream stream)
```

Comment:
I would suggest writing this a bit different.
Instead of returning a string, return void.

The caller knows the stream type and will know how to get the buffer.
This will avoid the case when the Stream is not a MemoryStream and you have to copy the bytes.
It is most likely the caller wants the bytes, otherwise why pass the Stream?